python_modules/dagster/dagster/_utils/source_position.py

```diff
@@ -125,9 +125,23 @@ def source_error(
         )
 
 
-def _format_indented_error_msg(col: int, msg: str) -> str:
+def _format_indented_error_msg(
+    value_error: bool,
+    source_pos: SourcePosition,
+    source_line: str,
+    msg: str,
+) -> str:
     """Format an error message with a caret pointing to the column where the error occurred."""
-    return " " * (col - 1) + f"^ {msg}"
+    inset = source_pos.start.col
+    if value_error:
+        source_key_end_col = source_pos.end.col
+        source_line_end_col = len(source_line)
+
+        approx_middle = source_key_end_col + int((source_line_end_col - source_key_end_col) / 2)
+        if approx_middle > source_pos.start.col:  # sanity check we got a reasonable value
```

Comment:
is this really necessary / are there conceivable cases where this would happen? doesn't hurt anything (so feel free to keep it in) just seems like a bit of a random thing to check